tests/helper_networks.rs

```diff
@@ -147,13 +150,19 @@ fn spawn_helpers(
             command.preserved_fds(vec![socket.as_raw_fd()]);
             command.args(["--server-socket-fd", &socket.as_raw_fd().to_string()]);
 
-            command.spawn().unwrap().terminate_on_drop()
+            // something went wrong if command is terminated at this point.
+            let mut child = command.spawn().unwrap();
+            if let Ok(Some(status)) = child.try_wait() {
+                panic!("Helper binary terminated early with status = {status}");
+            }
+
+            child.terminate_on_drop()
         })
         .collect::<Vec<_>>()
 }
 
 fn test_network(https: bool) {
-    let dir = tempdir().unwrap();
+    let dir = TempDir::new(false); // set to true if tempdir needs to be preserved
```

Comment:
nit: set to true if tempdir needs to be preserved ALWAYS. I think we are preserving it in case of a fail anyways by this change. Also, can you point to where the folder with failed run details can be found?